lib/orphans.nix

```diff
@@ -0,0 +1,97 @@
+{ lib }:
+let
+  inherit (builtins)
+    attrValues
+    concatStringsSep
+    length
+    mapAttrs
+    tryEval
+    isAttrs
+  ;
+  inherit (lib)
+    any
+    flatten
+    isDerivation
+    optional
+    types
+    concatLists
+    mapAttrsToList
+  ;
+  isAttr = types.attrs.check;
+in
+rec {
+  /* Return a list of maintainers from a package, empty list if inaccessible.
+
+     Example:
+       listMaintainers pkgs.foo
+       => [ { name = "Foo Bar"; email = "name@domain.com"; github = "foo"; } ]
+       listMaintainers pkgs.bar
+       => [ ]
+       listMaintainers {}
+       => error: listMaintainers must receive a package
+  */
+  listMaintainers =
+  # Must be a package
+  p:
+    if isDerivation p then
```

Comment:
I think this check is unnecessary, since it limits the usefulness of this function. For example, without this check we could also use it on modules to check which ones doesn't have maintainers.